icu4c/source/i18n/unicode/dtitvinf.h

```diff
@@ -343,7 +343,7 @@ class U_I18N_API DateIntervalInfo U_FINAL : public UObject {
     /**
      * Internal struct used to load resource bundle data.
      */
-    struct DateIntervalSink;
+    struct U_HIDDEN DateIntervalSink;
```

Comment:
Why is this important? It's a forward declaration in the private section of the class.
Same for U_HIDDEN in the other classes.